### ISSUE TITLE
Crash from UsdStage::~UsdStage()

### ISSUE THREAD

**Issue by zhoub** [PixarAnimationStudios/USD#1177]
### Steps to Reproduce
1.  Create massive UsdStage by TBB in C++ and delete them in parallel.
2. Sometimes it will crash with following stack info

```
  4   libsystem_kernel.dylib              0x00007fff6eaae7fa __pthread_kill + 10
  5   libsystem_c.dylib                   0x00007fff6ea35a1c abort + 120
  6   libtbb.dylib       0x00000001433dd40f tbb::assertion_failure(char const*, int, char const*, char const*) + 143
  7   libtbb.dylib       0x00000001433dffb3 tbb::task_scheduler_init::terminate() + 67
  8   libusd_ms.dylib                   0x000000015589b4d7
pxr_v0_19__pxrReserved__::ResetTBBTaskSchedInit() + 23
  9   libusd_ms.dylib                   0x0000000155e7d132 pxr_v0_19__pxrReserved__::UsdStage::~UsdStage() + 194
  10  libusd_ms.dylib                   0x0000000155e7de5e pxr_v0_19__pxrReserved__::UsdStage::~UsdStage() + 14
```

Back to the source file threadLimits.cpp
```cpp
tbb::task_scheduler_init *_tbbTaskSchedInit;

void ResetTBBTaskSchedInit(){
    if(_tbbTaskSchedInit) {
        _tbbTaskSchedInit->terminate(); // Here crashes from TBB.
        delete _tbbTaskSchedInit;
        _tbbTaskSchedInit = NULL;
    }
}

UsdStage::~UsdStage()
{
    TF_DEBUG(USD_STAGE_LIFETIMES).Msg(
        "UsdStage::~UsdStage(rootLayer=@%s@, sessionLayer=@%s@)\n",
        _rootLayer ? _rootLayer->GetIdentifier().c_str() : "<null>",
        _sessionLayer ? _sessionLayer->GetIdentifier().c_str() : "<null>");
    _Close();
    if (_mallocTagID != _dormantMallocTagID){
        free(const_cast<char*>(_mallocTagID));
    }

    DeleteTheArenaManager();
    ResetTBBTaskSchedInit();
}
```

So in massive threading mode, the variable *_tbbTaskSchedInit* should be protected.

Please have a look at this. Thanks.

### System Information (OS, Hardware)
Any OS.

### Package Versions
19.06

### Build Flags
Clang Release mode

**Comment by zhoub** [comment]
Testing latest 20.05, if this still happens will create a new issue.
